src/parameter.rs

```diff
@@ -13,6 +13,43 @@ pub(crate) struct Parameter<'src> {
   pub(crate) name: Name<'src>,
 }
 
+impl<'src> Parameter<'src> {
+  pub(crate) fn format_eval<'this, 'eval, 'run>(
+    &'this self,
+    evaluator: &'eval std::cell::RefCell<Evaluator<'src, 'run>>,
+  ) -> FmtEval<'this, 'eval, 'src, 'run> {
+    FmtEval(self, evaluator)
+  }
+}
+
+pub struct FmtEval<'param, 'eval, 'src, 'run>(
+  &'param Parameter<'src>,
+  &'eval std::cell::RefCell<Evaluator<'src, 'run>>,
+);
```

Comment:
I think even if the evaluator was included in `Wrapper`, it would probably cause major problems from the lifetimes alone.
I tried to use `... -> impl ColorDisplay + 'this`, even with explicit bounds that all other lifetimes outlive `'this`, it would complain about some lifetimes missing from the signature.

Not only that, but I imagine this would complicate the current approach of `Wrapper` which uses a trait object and probably faces similar issues.